libkineto/src/output_json.cpp

```diff
@@ -38,6 +38,37 @@ void ChromeTraceLogger::handleTraceStart(
   "schemaVersion": {},
   )JSON", kSchemaVersion);
 
+#ifdef HAS_CUPTI
+  const std::vector<cudaOccDeviceProp>& occProps = KINETO_NAMESPACE::occDeviceProps();
+  if (occProps.size() > 0) {
+    std::ostringstream oss;
+    oss << "[";
+    for (size_t i = 0; i < occProps.size(); i += 1) {
+      const cudaOccDeviceProp& occProp = occProps[i];
+      if (i > 0) {
+        oss << ", ";
+      }
+      oss << "{";
+      oss << "\"computeMajor\": " << occProp.computeMajor << ", ";
+      oss << "\"computeMinor\": " << occProp.computeMinor << ", ";
+      oss << "\"maxThreadsPerBlock\": " << occProp.maxThreadsPerBlock << ", ";
+      oss << "\"maxThreadsPerMultiprocessor\": " << occProp.maxThreadsPerMultiprocessor << ", ";
+      oss << "\"regsPerBlock\": " << occProp.regsPerBlock << ", ";
+      oss << "\"regsPerMultiprocessor\": " << occProp.regsPerMultiprocessor << ", ";
+      oss << "\"warpSize\": " << occProp.warpSize << ", ";
+      oss << "\"sharedMemPerBlock\": " << occProp.sharedMemPerBlock << ", ";
+      oss << "\"sharedMemPerMultiprocessor\": " << occProp.sharedMemPerMultiprocessor << ", ";
```

Comment:
need we consolidate with distributed training metadata like "id", "name" and "total_memory"?